Add actuator metrics support for r2dbc-pool

ttddyy opened the pull request:
Add "ConnectionPoolMetrics" meter binder and
"ConnectionPoolMetricsAutoConfiguration".

[resolves #17]

mp911de commented:
This looks pretty decent. Do you want to also adapt our `spring-boot-example-h2` and create a dedicated `spring-boot-example-r2dbc-actuator` that demonstrates the Micrometer integration?

ttddyy commented:
Ok, I have rebased and added a commit that adds `spring-boot-example-r2dbc-actuator` module.
The new module is basically a copy of `spring-boot-example-h2` and just changed `application.properties` and `pom.xml`.

mp911de commented:
Thanks a lot. That's merged and polished now.
